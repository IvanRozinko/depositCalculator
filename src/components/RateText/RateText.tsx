import React from 'react'
import { Typography } from '@material-ui/core';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles'; 

const useStyle = makeStyles((theme: Theme) => 
  createStyles({
    rate: {
      width: 100,
      fontSize: 20,
      fontWeight: 800,
      minWidth: '60px',
      marginBottom: 20,
      textAlign: 'center',
      alignSelf: 'center',
      [theme.breakpoints.down('xs')]: {
        marginBottom: 0
      }
    }
  })
);

interface RateTextProps {
  rate: string
}

function RateText({ rate }: RateTextProps) {
  const styles = useStyle();
  return (
    <Typography className={styles.rate}>
      {rate}
    </Typography>
  )
}

export default RateText
