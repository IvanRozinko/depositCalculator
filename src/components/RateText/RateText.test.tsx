import RateText from "./RateText";
import { shallow } from 'enzyme';

describe('<RateText /> :', () => {
  const rate = '10';

  const component = shallow(
    <RateText rate={rate} />
  )

  it('should renders correctly', () => {
    expect(component).toMatchSnapshot();
  })

  it('should display correct prop', () => {
    expect(component.text()).toBe(rate);
  })
})