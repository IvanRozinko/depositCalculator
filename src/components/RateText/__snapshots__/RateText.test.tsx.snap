// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RateText />  should renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-rate-1"
>
  10
</WithStyles(ForwardRef(Typography))>
`;

exports[`<RateText /> : should renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-rate-1"
>
  10
</WithStyles(ForwardRef(Typography))>
`;

exports[`<RateText /> should renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-rate-1"
>
  10
</WithStyles(ForwardRef(Typography))>
`;

exports[`<RateText should renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  className="makeStyles-rate-1"
>
  10
</WithStyles(ForwardRef(Typography))>
`;
